content-services/content-sources-backend
# Fixes 3599: Add candlepin content for templates

## Summary
Adds candlepin product/pool/environment/content creation on template create. If updating a template, content is removed from environments if necessary.

## Testing steps
1. Create  a custom repository and a red hat repository.
2. Create a content template with these repositories included. Wait for task to finish.
3. The will be many things to check in candlepin: the new product created, the new pool created, the new environment created for the template, the new product content, the new content promoted to the environment.

Here is an example to list the products. Make sure you include the Authorization header. It is the base64 encoded username:password (admin:admin by default).
```
GET http://localhost:8181/candlepin/owners/content-sources-test/products
Accept: application/json
Authorization: Basic YWRtaW46YWRtaW4=
```
4. Refer to the [candlepin REST API spec](https://petstore.swagger.io/?url=https://raw.githubusercontent.com/candlepin/candlepin/main/api/candlepin-api-spec.yaml#/) to see how to query more endpoints.
5. Make sure the content URLs and labels are set correctly. Keep in mind, these will be different for custom vs Red Hat repositories.
6. Check the content URL against the base URL of the pulp distribution associated to the repository in the template.
7. Update the template to remove a repository. Wait for the task to finish.
8. The content set for that repository should no longer be in the environment.
9. Reference the JIRA card to make sure you're checking all the things.

## Checklist
